src/components/Entry.vue

```diff
@@ -80,43 +79,60 @@
       class="bg-primary hover:bg-primary-hover text-white font-bold shadow-xl uppercase text-lg mx-auto p-4 mt-4 rounded disabled:opacity-50 btn"
       :disabled="isSubmitButtonDisabled"
     >
-      SUBMIT / जमा करें
+      {{ submitButtonDisplayText }}
     </button>
   </div>
 </template>
 
 <script>
-import { validateSRN } from "@/services/validation.js";
+import { validateID } from "@/services/validation.js";
 import { redirectToDestination } from "@/services/redirectToDestination.js";
 import { sendSQSMessage } from "@/services/API/sqs";
-
-const NUMBER_OF_INPUTS_ALLOWED = 10;
-const authType = "SRN";
+import { validationTypeToFunctionMap } from "@/services/basicValidationMapping.js";
 
 export default {
-  name: "SRNEntry",
+  name: "Entry",
   props: {
     redirectTo: String,
     redirectID: String,
     purpose: String,
     purposeParams: String,
+    programData: Object,
+    program: String,
+    authType: String,
   },
   data() {
     return {
       userIDList: [{ userID: "", valid: false }], //array containing user-ids and a valid flag for each
-      invalidInputMessage: null, // whether the input is in correct format
       isCurrentUserValid: false, // whether the current user is valid
-      maxLengthOfSRN: 10,
       validateCount: 0, // count the number of times the user has been validated
-      invalidLoginMessage: "Please enter correct SRN / कृपया सही SRN दर्ज करें",
+      invalidLoginMessage: "",
       isLoading: false,
+      invalidInputMessage: null, // whether the input being entered by the user matches the basic validation criteria
     };
   },
+
   computed: {
+    /** Returns the input mode stored against the program */
+    inputMode() {
+      return this.programData.input.mode;
+    },
+
+    /** Returns the input type stored against the program */
+    inputType() {
+      return this.programData.input.type;
+    },
+
+    /** Returns the placeholder text stored against the program */
+    inputBoxPlaceholderText() {
+      return this.programData.text.default.placeholder;
```

Comment:
1. Add error checking to make sure all required fields exist in the JSON
2. Add documentation to README explaining what the required fields are, and what are optional (if any)